OpenRA.Game/Traits/ActivityUtils.cs

```diff
@@ -44,7 +44,7 @@ public static Activity RunActivity(Actor self, Activity act)
 				else
 					start = current;
 
-				if (prev == act)
+				if (act == prev || act == prev.Parent)
```

Comment:
What you are missing is that it would make both types of activities look and behave differently to the outside world. Remember that `CompositeActivity` is a kind-of `Activity`, so it needs to act like that. If you add a new property instead of overriding `NextActivity`, code like [this](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/AI/States/AirStates.cs#L136) might break unless you check what kind of activity you're dealing with. The way I wrote this, it's all transparent to users and they don't need to care at all (yes, I realize that's kind of what you're complaining about).